src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -245,26 +249,66 @@ private void preValidatePostalContactDetails(List<Product> products, Contact con
   }
 
   private void recordRateLimiting(
-      Contact contact, String ipAddress, List<Product> products, CollectionCase caseDetails)
-      throws CTPException {
-
+      Contact contact, String ipAddress, List<Product> products, CollectionCase caseDetails) {
     if (appConfig.getRateLimiter().isEnabled()) {
       for (Product product : products) {
         log.with("fulfilmentCode", product.getFulfilmentCode()).debug("Recording rate-limiting");
         CaseType caseType = CaseType.valueOf(caseDetails.getCaseType());
         UniquePropertyReferenceNumber uprn =
             UniquePropertyReferenceNumber.create(caseDetails.getAddress().getUprn());
-
-        // if the limit is breached, a ResponseStatusException, typically with 429 HTTP code
-        // will be thrown.
-        rateLimiterClient.checkRateLimit(
-            Domain.RH, product, caseType, ipAddress, uprn, contact.getTelNo());
+        recordRateLimiting(contact, product, caseType, ipAddress, uprn);
       }
     } else {
       log.info("Rate limiter client is disabled");
     }
   }
 
+  /*
+   * Call the rate limiter within a circuit-breaker, thus protecting the RHSvc
+   * functionality from the unlikely event that that the rate limiter service is failing.
+   *
+   * If the limit is breached, a ResponseStatusException with HTTP 429 will be thrown.
+   */
+  private void recordRateLimiting(
+      Contact contact,
+      Product product,
+      CaseType caseType,
+      String ipAddress,
+      UniquePropertyReferenceNumber uprn) {
+    ResponseStatusException limitException =
+        circuitBreaker.run(
+            () -> {
+              try {
+                rateLimiterClient.checkRateLimit(
+                    Domain.RH, product, caseType, ipAddress, uprn, contact.getTelNo());
+                return null;
+              } catch (CTPException e) {
+                // we should get here if the rate-limiter is failing or not communicating
+                // ... wrap and rethrow to be handled by the circuit-breaker
+                throw new RuntimeException(e);
+              } catch (ResponseStatusException e) {
+                // we have got a 429 but don't rethrow it otherwise this will count against
+                // the circuit-breaker accounting, so instead we return it to later throw
+                // outside the circuit-breaker mechanism.
+                return e;
+              }
+            },
+            throwable -> {
```

Comment:
The Spring javadoc doesn't describe what the fallback block is for or when it's used, so I think that a 1 line comment here would help. ie. just summarising the circumstances in which the block is executed.

Reply:
added some additional comments